impl_core/src/main/java/io/opencensus/implcore/trace/propagation/BinaryFormatImpl.java

```diff
@@ -96,25 +106,36 @@ public SpanContext fromByteArray(byte[] bytes) throws SpanContextParseException
     if (bytes.length == 0 || bytes[0] != VERSION_ID) {
       throw new SpanContextParseException("Unsupported version.");
     }
-    TraceId traceId = TraceId.INVALID;
-    SpanId spanId = SpanId.INVALID;
+    if (bytes.length < FORMAT_LENGTH - ID_SIZE - TraceOptions.SIZE) {
+      throw new SpanContextParseException("Invalid input: truncated");
+    }
+    TraceId traceId;
+    SpanId spanId;
     TraceOptions traceOptions = TraceOptions.DEFAULT;
     int pos = 1;
-    try {
-      if (bytes.length > pos && bytes[pos] == TRACE_ID_FIELD_ID) {
-        traceId = TraceId.fromBytes(bytes, pos + ID_SIZE);
-        pos += ID_SIZE + TraceId.SIZE;
-      }
-      if (bytes.length > pos && bytes[pos] == SPAN_ID_FIELD_ID) {
-        spanId = SpanId.fromBytes(bytes, pos + ID_SIZE);
-        pos += ID_SIZE + SpanId.SIZE;
+    if (bytes[pos] == TRACE_ID_FIELD_ID) {
+      traceId = TraceId.fromBytes(bytes, pos + ID_SIZE);
+      pos += ID_SIZE + TraceId.SIZE;
+    } else {
+      throw new SpanContextParseException("Invalid input: expected trace ID at offset " + pos);
+    }
+    if (bytes[pos] == SPAN_ID_FIELD_ID) {
+      spanId = SpanId.fromBytes(bytes, pos + ID_SIZE);
+      pos += ID_SIZE + SpanId.SIZE;
+    } else {
+      throw new SpanContextParseException("Invalid input: expected span ID at offset " + pos);
+    }
+    // The trace options field is optional. However, when present, it should be valid.
+    if (bytes.length > pos) {
+      if (bytes[pos] != TRACE_OPTION_FIELD_ID) {
+        throw new SpanContextParseException(
```

Comment:
I would not throw an exception in this case. If in the future we want to extend the fields this let's us do that. What do you think?